industrial_ci/src/tests/source_tests.sh

```diff
@@ -158,6 +158,12 @@ if [ "$BUILDER" == catkin ]; then catkin build $OPT_VI --summarize  --no-status
 
 ici_time_end  # catkin_build
 
+if [ "$CATKIN_LINT" == "true" ]; then
+    ici_time_start catkin_lint
+    catkin_lint --strict $CATKIN_LINT_ARGS $CATKIN_WORKSPACE/src && echo "catkin_lint passed." || error "`catkin_lint --strict` failed by either/both errors and/or warnings"
```

Comment:
IMHO the echo ouput are superfluous, the fold will highlight the error anyway.

Reply:
If you check `$CATKIN_WORKSPACE/src`, the rosinstall will get included.
It ight be better to check only `"$CATKIN_WORKSPACE/src/$(basename $TARGET_REPO_PATH)"`

Reply:
Or simply `"$TARGET_REPO_PATH"`.

Reply:
> the echo ouputs are superfluous, the fold will highlight the error anyway.

`echo` is called only upon success, so no problem?

Reply:
It's is harder to read / maintain and the ouput does not add any information
So I would remove it, but I leave the decision to you.

Reply:
I found the output of `catkin_lint` is often not clearly telling me if it passed or not, e.g. with the following that includes a line from my commit (`catkin_lint passed.`), if you're not that used to the tool then it could be hard to tell without my commit. With folded in green-color, most of the cases you might not even to unfold the section, but I'd still prefer to being explicit.

https://travis-ci.org/ros-industrial/industrial_ci/jobs/289293486
```
catkin_lint: checked 1 packages and found 1 problems
catkin_lint: 1 notices have been ignored. Use -W2 to see them
catkin_lint passed.
<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<<
Function catkin_lint returned with code '0' after 0 min 3 sec 
```

> It's harder to read / maintain 

How so?

Reply:
> I found the output of catkin_lint is often not clearly telling me if it passed or not

The state if encoded in "Function catkin_lint returned with code '0' after 0 min 3 sec" and the color.

>> It's harder to read / maintain

>How so?

`catkin_lint --strict $CATKIN_LINT_ARGS $CATKIN_WORKSPACE/src && echo "catkin_lint passed." || error "catkin_lint --strict failed by either/both errors and/or warnings"` 
(Beware: do not use the backticks in the string, it is the same as `$()`!)

vs.

`catkin_lint --strict $CATKIN_LINT_ARGS $CATKIN_WORKSPACE/src`